src/App.tsx

```diff
@@ -15,49 +16,121 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType !== SortType.NONE) {
+    if (sortType === SortType.ALPHABET) {
+      visibleGoods.sort((g1, g2) => {
+        return g1.localeCompare(g2);
+      });
+    }
+
+    if (sortType === SortType.LENGTH) {
+      visibleGoods.sort((g1, g2) => {
+        return g1.length - g2.length;
+      });
+    }
+  }
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const alphabet = () => setSortType(SortType.ALPHABET);
+
+  const length = () => setSortType(SortType.LENGTH);
+
+  const reverse = () => setReversed(!isReversed);
```

Comment:
if you set state based on previous state, which you do, use functional set state